client/my-sites/stats/activity-log/index.jsx

```diff
@@ -122,38 +131,49 @@ class ActivityLog extends Component {
 		this.handlePeriodChange( ...args );
 	};
 
-	handleRequestRestore = ( requestedRestoreTimestamp, from ) => {
-		this.props.recordTracksEvent( 'calypso_activitylog_restore_request', {
+	handleRequestRestore = ( activityId, from ) => {
+		const { recordTracksEvent, rewindRequestRestore, siteId } = this.props;
+
+		recordTracksEvent( 'calypso_activitylog_restore_request', {
 			from,
-			timestamp: requestedRestoreTimestamp,
-		} );
-		this.setState( {
-			requestedRestoreTimestamp,
-			showRestoreConfirmDialog: true,
+			activityId,
 		} );
+		rewindRequestRestore( siteId, activityId );
 	};
 
 	handleRestoreDialogClose = () => {
-		this.props.recordTracksEvent( 'calypso_activitylog_restore_cancel', {
-			timestamp: this.state.requestedRestoreTimestamp,
+		const {
+			recordTracksEvent,
+			requestedRestoreActivityId,
+			rewindRequestDismiss,
+			siteId,
+		} = this.props;
+		recordTracksEvent( 'calypso_activitylog_restore_cancel', {
+			activityId: requestedRestoreActivityId,
 		} );
-		this.setState( { showRestoreConfirmDialog: false } );
+		rewindRequestDismiss( siteId );
 	};
 
 	handleRestoreDialogConfirm = () => {
-		const { recordTracksEvent, rewindRestore, siteId } = this.props;
-		const { requestedRestoreTimestamp } = this.state;
+		const {
+			recordTracksEvent,
+			requestedRestoreActivity,
+			requestedRestoreActivityId,
+			rewindRestore,
+			siteId,
+		} = this.props;
+		const { activityTs } = requestedRestoreActivity;
 
 		recordTracksEvent( 'calypso_activitylog_restore_confirm', {
-			timestamp: requestedRestoreTimestamp,
+			activityId: requestedRestoreActivityId,
```

Comment:
Is it ok to change the content of existing tracks events like this? Probably fine since we won't have seen much usage yet, but an alternative would be to introduce a new event and retire the old one.

Reply:
Thanks, good point! This can be handled in the event validation, maybe a good motivation to prioritize that 😅 

see 174-gh-tracks-events-validation-status

Reply:
After some though, including an `activityId` likely adds no value to tracks.

We can still track the timestamp when the user _confirms_ the requested restore, which is probably plenty of information.